learners/BF_measles.Rmd

```diff
@@ -0,0 +1,884 @@
+transmission rate $( \beta )$. This reduction can be achieved through
+measures such as quarantine, limiting public gatherings, and regular
+screening in healthcare settings. To include such interventions in the
+baseline SEIR model, the `{epidemics}` package provides an object called
+`interventions`, which is added to the model definition as shown in the
+code chunk below.
+
+
+```{r}
+reduced_contacts <- epidemics::intervention(
+  name = "reduce",
+  type = "rate",
+  time_begin = 30,
+  time_end = 30 + 60,
+  reduction = 0.2
+)
+
+# Run the epidemic model
+output2 <- epidemics::model_default(
+  population = bf_population,
+  transmission_rate = 15 / 5,  # R0/infectious period = 15/5 = 3
+  infectiousness_rate = 1 / 8,  # 1/latent period (pre-infectious phase)
+  recovery_rate = 1 / 5,  # 1/infectious period
+  intervention = list(transmission_rate = reduced_contacts),
+  time_end = 120,  # Simulation over 75 days
+  increment = 1  # Daily increments
+)
+```
+
+
+To examine the effect that implementing a reduction in contact has on
+the daily number of new infections, we plot the results from the two
+models.
+
+
+```{r}
+# Load new infections data for each scenario
+data_baseline <- epidemics::new_infections(output, by_group = TRUE)
+data_reduced <- epidemics::new_infections(output2, by_group = TRUE)
+
+# Assign scenario names
+data_baseline$scenario <- "Baseline"
+data_reduced$scenario <- "Reduced Contact"
+
+# Combine the data from both scenarios
+data_combined <- bind_rows(data_baseline, data_reduced)
+
+# Plot new infections over time with scenario and demographic groups
+ggplot(data_combined) +
+  geom_line(
+    aes(
+      x = time, y = new_infections,
+      col = demography_group, linetype = scenario
+    )
+  ) +
+  # Highlight the reduced contact intervention period
+  annotate(
+    geom = "rect",
+    xmin = reduced_contacts[["time_begin"]],
+    xmax = reduced_contacts[["time_end"]],
+    ymin = 0, ymax = 0.67e3,
+    fill = alpha("red", 0.2),
+    linetype = "dashed"
+  ) +
+  # Set axis scaling and formatting
+  scale_y_continuous(labels = scales::comma) +
+  scale_linetype_manual(
+    name = "Scenario",
+    values = c(
+      "Baseline" = "dashed",
+      "Reduced Contact" = "solid"
+    )
+  ) +
+  scale_colour_brewer(
+    palette = "Dark2",
+    name = "Age Group"
+  ) +
+  expand_limits(y = c(0, 1e3)) +
+  coord_cartesian(expand = FALSE) +
+  theme_bw() +
+  theme(
+    legend.position = "top"
+  ) +
+  labs(
+    x = "Simulation Time (days)",
+    y = "New Infections",
+    linetype = "Scenario",
+    caption =
+      expression(
+        paste(
+          "New infections under baseline and 20% reduced transmission rate (",
+          beta,
+          ") for different age groups."
+        )
+      )
+  ) +
+  # Adjust legend display to fit in two rows
+  guides(
+    color = guide_legend(nrow = 2)
+  )
+```
+
+## Vaccination Model
+
+An effective vaccine for measles exists, known as the measles, mumps,
```

Comment:
Do all countries administer MMR? It's a common vaccine but maybe not universal?

Reply:
Yes, not universal. Burkina faso does administer MMR, now this clarification has been included.